shell/platform/common/cpp/accessibility_bridge.h

```diff
@@ -0,0 +1,240 @@
+  AccessibilityBridge(std::unique_ptr<AccessibilityBridgeDelegate> delegate,
+                      void* user_data);
+  ~AccessibilityBridge();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Adds a semantics node update to the pending semantics update.
+  ///             Calling this method alone will NOT update the semantics tree.
+  ///             To flush the pending updates, call the CommitUpdates().
+  ///
+  /// @param[in]  node           A pointer to the semantics node update.
+  void AddFlutterSemanticsNodeUpdate(const FlutterSemanticsNode* node);
+
+  //------------------------------------------------------------------------------
+  /// @brief      Adds a custom semantics action update to the pending semantics
+  ///             update. Calling this method alone will NOT update the
+  ///             semantics tree. To flush the pending updates, call the
+  ///             CommitUpdates().
+  ///
+  /// @param[in]  action           A pointer to the custom semantics action
+  /// update.
+  void AddFlutterSemanticsCustomActionUpdate(
+      const FlutterSemanticsCustomAction* action);
+
+  //------------------------------------------------------------------------------
+  /// @brief      Flushes the pending updates and applies them to this
+  ///             accessibility bridge.
+  void CommitUpdates();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the underlying AXTree.
+  AXTree* GetAXTree();
+
+  //------------------------------------------------------------------------------
+  /// @brief      The event generator of this accessibility bridge. It contains
+  ///             the pending accessibility events generated as a result of a
+  ///             semantics update.
+  AXEventGenerator* GetEventGenerator();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the user data.
+  void* GetUserData();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the accessibility bridge delegate.
+  AccessibilityBridgeDelegate* GetDelegate();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the flutter accessibility node with the given id from this
+  ///             accessibility bridge.
+  ///
+  /// @param[in]  id           The id of the flutter accessibility node you want
+  ///                          to retrieve.
+  FlutterAccessibility* GetFlutterAccessibilityFromID(int32_t id) const;
+
+  //------------------------------------------------------------------------------
+  /// @brief      Update the currently focused flutter accessibility node.
+  ///
+  /// @param[in]  id           The id of the currently focused flutter
+  ///                          accessibility node.
+  void SetFocusedNode(int32_t node_id);
```

Comment:
Using int32_t here but int16_t above for the same concept seems dangerous. We should typedef this.